File: py/longest_word/solution_test.py
import unittest
from solution import Solution

class TestLongestWord(unittest.TestCase):
    def setUp(self):
        self.solution = Solution()

    def test_case_one(self):
        words = "what a wonderful world"
        expected = "wonderful"
        result = self.solution.longest_word(words)
        self.assertEqual(result, expected)

    def test_case_two(self):
        words = "have a nice day"
        expected = "nice"
        result = self.solution.longest_word(words)
        self.assertEqual(result, expected)

    def test_case_three(self):
        words = "the quick brown fox jumped over the lazy  dog"
        expected = "jumped"
        result = self.solution.longest_word(words)
        self.assertEqual(result, expected)

    def test_case_four(self):
        words = "who did eat the ham"
        expected = "ham"
        result = self.solution.longest_word(words)
        self.assertEqual(result, expected)

    def test_case_five(self):
        words = "potato"
        expected = "potato"
        result = self.solution.longest_word(words)
        self.assertEqual(result, expected)

File: java/longestWord/src/test/java/dev/jgn/SolutionTest.java
package dev.jgn;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class SolutionTest {
    Solution solution = new Solution();

    @Test
    public void testCaseOne() {
        var sentence = "what a wonderful world";
        var expected = "wonderful";
        var result = solution.longestWord(sentence);

        assertEquals(expected, result);
    }

    @Test
    public void testCaseTwo() {
        var sentence = "have a nice day";
        var expected = "nice";
        var result = solution.longestWord(sentence);

        assertEquals(expected, result);
    }

    @Test
    public void testCaseThree() {
        var sentence = "the quick brown fox jumped over the lazy dog";
        var expected = "jumped";
        var result = solution.longestWord(sentence);

        assertEquals(expected, result);
    }

    @Test
    public void testCaseFour() {
        var sentence = "who did eat the ham";
        var expected = "ham";
        var result = solution.longestWord(sentence);

        assertEquals(expected, result);
    }

    @Test
    public void testCaseFive() {
        var sentence = "potato";
        var expected = "potato";
        var result = solution.longestWord(sentence);

        assertEquals(expected, result);
    }
}
